src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs

```diff
@@ -541,7 +552,9 @@ private void ExplicitExpression()
             using (PushSpanConfig(ConfigureExplicitExpressionSpan))
             {
                 var success = Balance(
-                    BalancingModes.BacktrackOnFailure | BalancingModes.NoErrorOnFailure | BalancingModes.AllowCommentsAndTemplates,
+                    BalancingModes.BacktrackOnFailure |
+                        BalancingModes.NoErrorOnFailure |
```

Comment:
Just indenting the rest of the long expression so the parts don't look like additional arguments.